# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1295746575 28800
#      Sat Jan 22 17:36:15 2011 -0800
# Node ID 2f728f9d32d488a1285c8dde911418d3db0ef3c5
# Parent  ddd4ef7934fcd963ebc6deba9e8f36aae12b0674
README.MinGW: Replace placeholder with actual instructions for platform.

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2010-01-22  Rik  <octave@nomad.inbox5.com>
+
+	* README.Windows: Reference README.MinGW
+	* README.MinGW: Replace placeholder with actual instructions
+
 2011-01-22  John W. Eaton  <jwe@octave.org>
 
 	* configure.ac (AC_INIT): Version is now 3.5.0+.
 	(OCTAVE_API_VERSION_NUMBER): Now 44.
 	(OCTAVE_RELEASE_DATE): Now 2011-01-22.
 	(OCTAVE_COPYRIGHT): Update year.
 
 2010-01-20  Rik  <octave@nomad.inbox5.com>
diff --git a/README.MinGW b/README.MinGW
--- a/README.MinGW
+++ b/README.MinGW
@@ -1,1 +1,108 @@
-Placeholder until file written by Tatsuro and reviewed.
+1. General Users
+================
+
+A MInGW bundle is available from sourceforge.
+
+ http://octave.sourceforge.net/index.html
+
+2. Developers
+=============
+
+The process for building Octave from source is as follows.
+
+ * Install development tools and compiler
+ * Install or compile dependencies of Octave
+ * Download Octave source
+ * Configure and compile source
+
+We strongly encourage experienced users to build Octave from the
+testing and development snapshots to try the latest features of Octave.
+
+A good tutorial on building Octave for version 3.2 is available from 
+http://octave.svn.sourceforge.net/viewvc/octave/trunk/octave-forge/admin/Windows/mingw32/HOWTO.txt 
+
+Additional information beyond the tutorial is summarized below.
+
+ 2.1 Development tools and compiler
+ ----------------------
+ Msys + MinGW development Tools
+
+ The Msys + MinGW development Tools are required to build Octave from source.
+
+ The latest version of the Msys system is 1.0.16.  For MinGW, search and use
+ the most recent version available.  Note: The Howto.txt file describing
+ the development tools is rather old.
+
+ The latest GCC (GNU Complier Collection) for MinGW is gcc-4.5.0.  Earlier
+ versions are not recommended for building Octave 3.4.xx.
+
+ The latest Msys, MinGW, and GCC tools can be downloaded from 
+ http://sourceforge.net/projects/mingw/files/ 
+
+ (Compiling for 64-bit binaries requires mingw-64 toolkits.  This has not been
+  tested.)
+
+ 2.2 Dependencies
+ ----------------------
+
+ External program dependencies
+ 
+  * gnuplot (you can build this with the shell script in Howto.txt)
+  * MikTeX
+  * ghostscript
+  * pstoedit (required for printing using FLTK (OpenGL) graphics)
+  * fig2dev (Optional; better quality for FLTK printing in some file formats)
+  * sed
+  * less
+
+ The tools below are required for building the development sources from
+ the Mercurial archive, but not for simply building Octave from a snapshot
+ tar.gz file.
+
+  * Mercurial
+  * git
+  * automake
+  * autoconf
+
+ Library dependencies
+
+  Please read Howto.txt carefully on how to build them.  The Howto.txt is ols
+  so some revisions will be required when using GCC-4.5.0
+
+ 2.3 Download Octave source
+ ----------------------
+
+  The source for testing snapshots can be downloaded from the official Octave
+  web site.  http://www.gnu.org/software/octave/download.html 
+
+  The very latest development code is also available via anonymous access to a
+  read-only Mercurial archive.  The method of download is described on the web
+  site.  You will also need to install gnulib as another dependency.  Please
+  visit http://www.gnu.org/software/gnulib/ and download sources using git.
+
+ 2.4 Configure and compile source
+ ----------------------
+
+  Building from source can be carried out using the script described in
+  Howto.txt.  
+  
+  When using GCC-4.5.0, remove flags for shared libstdc++ because a shared
+  build is already the default.
+
+  When running './configure', you must add '--enable-float-truncate' option. 
+  Please see the following thread for more information: 
+  http://thread.gmane.org/gmane.comp.gnu.octave.bugs/12361/focus=12404
+
+  To build development sources from the Mercurial repository you must run
+  'autogen.sh' (bootstrap) before running configure and make.  This requires
+  that automake and autoconf tools be installed.
+
+
+Tatsuro MATSUOKA
+tmacchant@yahoo.co.jp
+
+Benjamin Lindner
+bjmldn@gmail.com 
+
+
+Last updated: Sat Jan 22 17:29:12 PST 2011
diff --git a/README.Windows b/README.Windows
--- a/README.Windows
+++ b/README.Windows
@@ -1,11 +1,11 @@
 See the file README.Cygwin for information about installing the binary
 package of Octave for Cygwin.
 
-See the file README.MinGW for information about installing the binary
-package of Octave for MinGW.
+See the file README.MinGW for information about Octave on the MinGW platform
+including installation of binaries or compiling Octave frome source.
 
 
 John W. Eaton
 jwe@octave.org
 
-Last updated: Thu Jan 20 10:35:23 PST 2011
+Last updated: Sat Jan 22 16:56:05 PST 2011
